wazuh/wazuh-kibana-app
# Create service to restart wazuh

**Description:**

The way to restart the manager or cluster was centralized in a service since each view that had the option to restart did it differently.


In order to allow the cluster to synchronize ruleset changes, a delay was applied before restarting Wazuh. This delay has been removed as it's been replaced by a polling system, so it's no longer required. The app will ping the Wazuh API and wait until it's back up again. If this does not happen, the app will show an error message, as the API did not come back online.

Due to the changes of this development, code has been removed as it was no longer used.

PR above with more information (a revert was made):

- #4365

**Issue:**

- #4277 

**Test:**

To test it you have to go to any page that has the restart option and try doing the restart, when it tells you that it has restarted try navigating between pages that make requests to the API.

_Successful restart_

1. Navigate to one of the following sections:
 - Management > Configuration > Edit configuration 
 - Management > Status
 - Management > Rules and edit file 
 - Management > Decoders and edit file 
 - Management > CDB lists and edit file 
2. Click on the restart button and wait until it's done.
3. Navigate to another route that uses some request for the API, and check that no error is thrown.

_Restart failure_

1. Navigate to one of the following sections:
- Management > Configuration > Edit configuration
- Management > Status
- Management > Rules and edit file 
- Management > Decoders and edit file 
- Management > CDB lists and edit file 
2. Click on the restart button and wait until it's done.
3. Once the first polling cycle is done, disconnect the manager from the app (in Docker, stop the container).
4. Wait until the modal changes state, showing that the restart has failed. A button to redirect to the Healthcheck must also appear. Check this button works properly.

**Screenshot:**

Ongoing restart:

![image](https://user-images.githubusercontent.com/63758389/187881860-f6142e4f-d896-448e-85cb-fcda55e46e45.png)

Successfully restarted:

![image](https://user-images.githubusercontent.com/63758389/187884410-b595e296-6e81-4131-8af0-734fd0f907ae.png)

Restart error:

![image](https://user-images.githubusercontent.com/63758389/187882086-3251b5fd-5b10-4ab4-a536-46bf54fd8961.png)

Cluster synchronization:

![image](https://user-images.githubusercontent.com/63758389/187882272-25404ebc-9b0b-4838-b833-65954db5e372.png)

Cluster synchronization error:

![image](https://user-images.githubusercontent.com/63758389/187882379-9cf37535-e7ea-4b7f-832f-bb2cf4ddd015.png)
